### DISCUSSION TITLE
Change in extension name command after 1.96 insider version.

### DISCUSSION THREAD

**Discussion by c3-rodolfobojorquez** [microsoft/vscode-discussions#2420] [Extension Development QnA] [answered]
I have been working on testing the extension from my company, one of our automated test case uses the name of a command that now is modified without any explanation:

Before:
`C3 AI: Focus on C3 AI View`
After:
![image](https://github.com/user-attachments/assets/b6d8cdbe-62fd-4dec-81ef-7562dcfaafc3)

**Comment by alefragnani** [comment, marked as answer]
Hi @c3-rodolfobojorquez ,

I didn’t notice such behavior on any of my extensions, nor find any issue reported on VS Code repo. 

Are you sure your company team didn’t change anything on the extension? I mean, the `C3 AI` _prefix_ may be defined as part of the command `title`, or can be concatenated from the command `category`. And both properties may be _located_ and defined on external resources. Maybe the team changed something without warning you.

```json
{
    “commands": [
            {
                "command": "bookmarks.toggle",
                "title": "%bookmarks.commands.toggle.title%",
                "category": "%bookmarks.commands.category.bookmarks%"
            },
}
```

Package.nls.json

```json
{
    "bookmarks.commands.category.bookmarks": "Bookmarks",
    "bookmarks.commands.category.bookmarks.selection": "Bookmarks (Selection)",
    "bookmarks.commands.toggle.title": "Toggle",
}
```

Hope this helps

**Comment by c3danielxu** [comment]
Hello @alefragnani @isidorn, thank you for responding! I work on the C3 AI extension and was able to create a reproducible scenario. All `Focus on ____ View` commands are provided by VS Code for any `WebviewViewProvider`s that are registered by an extension. I created a dummy webview, and in VS Code 1.95.3 it looks like this: `<activity bar title>: Focus on <webview name> View`

<img width="1493" alt="Screenshot 2024-12-03 at 10 57 22 AM" src="https://github.com/user-attachments/assets/c01f9b39-b614-46f1-bb5f-9dda14120523">

In VS Code Insiders (1.96) it looks like this: `Focus on <webview name> View`:
<img width="1496" alt="Screenshot 2024-12-03 at 11 01 31 AM" src="https://github.com/user-attachments/assets/a1404f0a-be52-42b7-b809-ad54938c8a36">

The activity bar title is missing in insiders version. Repro case below:

`package.json` of the extension:

```json
{
  ... // insert extension name, description, etc...
  "contributes": {
    "viewsContainers": {
      "activitybar": [
        {
          "id": "activitybar",
          "title": "Custom Activity Bar",
          "icon": "insert path to icon image"
        }
      ]
    },
    "views": {
      "activitybar": [
        {
          "type": "webview",
          "id": "customWebview",
          "name": "My Custom Webview"
        }
      ]
    },
```

`extension.ts`:
```js
export class CustomWebview implements WebviewViewProvider {
  // Called when a view first becomes visible.
  public resolveWebviewView(
    webviewView: WebviewView,
    context: WebviewViewResolveContext<unknown>,
    token: CancellationToken,
  ): void | Thenable<void> {
    webviewView.webview.html = `<!DOCTYPE html>
<html>
<head>
    <title></title>
</head>
<body>
    <p>Hello World!</p>
</body>
</html>`;
  }
}

export async function activate(context: ExtensionContext) {
  context.subscriptions.push(
    window.registerWebviewViewProvider('customWebview', new CustomWebview(), {
      webviewOptions: {
        retainContextWhenHidden: true,
      },
    }),
  );
}
```

**Reply by c3danielxu** [reply]
@alefragnani @isidorn @c3-rodolfobojorquez Since this seems like a bug, I have created an issue in the `vscode` repository: https://github.com/microsoft/vscode/issues/235220

Please let me know if there's any additional information needed.
